qiskit/circuit/quantumcircuit.py

```diff
@@ -3307,6 +3307,9 @@ def depth(
     ) -> int:
         """Return circuit depth (i.e., length of critical path).
 
+        .. warning::
+            This operation is not well defined if the circuit contains control-flow operations.
```

Comment:
would it make sense to add more info in this warning? would it be inaccurate to say, for example:

```suggestion
            This operation is not well defined if the circuit contains control-flow operations, but 
            their variables and clbits are counted in the depth calculation.
            
```

Reply:
I was mostly deliberately trying to avoid saying _anything_ about what happens in the return value of this if there's control-flow operations, since the whole result is ill defined.  I didn't want to commit to anything one way or the other.

Reply:
I suspected that was the case, and I think that makes sense too.